Framework/DataHandling/src/GroupDetectors2.cpp

```diff
@@ -668,14 +669,14 @@ void GroupDetectors2::processMatrixWorkspace(
 
     // translate to detectors
     std::vector<detid_t> det_ids;
-    try {
-      const Geometry::IDetector_const_sptr det = groupWS->getDetector(i);
+    if (spectrumInfo.hasDetectors(i)) {
+      const auto &det = spectrumInfo.detector(i);
 
-      Geometry::DetectorGroup_const_sptr detGroup =
-          boost::dynamic_pointer_cast<const Geometry::DetectorGroup>(det);
+      try {
+        const Geometry::DetectorGroup &detGroup =
```

Comment:
It might be better to use `SpectrumInfo`'s `SpectrumDefinition` for this. That depends on Simon's PR. But again would clean up a lot of this older `Geometry` code.